TombEngine/Game/control/box.cpp

```diff
@@ -878,18 +888,18 @@ int CreatureCreature(short itemNumber)
 	return 0;
 }
 
-int ValidBox(ItemInfo* item, short zoneNumber, short boxNumber) 
+bool ValidBox(ItemInfo* item, short zoneNumber, short boxNumber) 
 {
 	if (boxNumber == NO_BOX)
 		return false;
-
 	auto* creature = GetCreatureInfo(item);
-	int* zone = g_Level.Zones[(int)creature->LOT.Zone][FlipStatus].data();
-	if (creature->LOT.Fly == NO_FLYING && zone[boxNumber] != zoneNumber)
+	auto* zone = g_Level.Zones[(int)creature->LOT.Zone][FlipStatus].data();
+
+	if ((creature->LOT.Fly == NO_FLYING && zone[boxNumber] != zoneNumber) || boxNumber == NO_BOX)
```

Comment:
Why extra `boxNumber == NO_BOX` check is added here? It already exists in the very beginning of this function.

Reply:
Fixed